Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -2141,6 +2201,50 @@ def save_channels_to_context(updated_channel_list):
     set_to_integration_context_with_retries({'conversations': conversations}, OBJECTS_TO_KEYS, SYNC_CONTEXT)
 
 
+def search_context_for_user(user_name: str):
+    """
+    Searches the integration context to find a user ID that matches the given user's name
+
+    Args:
+        user_name: The user's name
+
+    Returns:
+        The slack user
+    """
+    integration_context = get_integration_context(SYNC_CONTEXT)
+
+    # Find user in the cache
+    users = integration_context.get('users')
+    if users:
+        users = json.loads(users)
+        user_to_search = user_name.lower()
+        users_filter = list(filter(lambda u: u.get('name', '').lower() == user_to_search
+                                             or u.get('profile', {}).get('email', '').lower() == user_to_search
+                                             or u.get('real_name', '').lower() == user_to_search, users))
+        if users_filter:
+            return users_filter[0]
+    else:
+        return None
+
+
+def save_users_to_context(updated_users_list):
+    """
+    Saves a list of users into the integration context.
+
+    Args:
+        updated_users_list: A list of users
+    """
+    # Save conversations to cache
+    integration_context = get_integration_context(SYNC_CONTEXT)
+    users = integration_context.get('users')
+    if users:
+        users = json.loads(users)
+        users.extend(updated_users_list)
```

Comment:
this can create duplicates in the list. we need to rethink if users should be a dict, or you should have on function that adds one user and another function that replaces all the users after you downloaded the complete new list. (maybe for replacing you dont even need a separated function)